src/ode_solver/ode_solver_factory.cpp

```diff
@@ -146,38 +147,61 @@ std::shared_ptr<ODESolverBase<dim,real,MeshType>> ODESolverFactory<dim,real,Mesh
         }
     }
     if (ode_solver_type == ODEEnum::rrk_explicit_solver){
-        if constexpr(dim==1){
-            const bool use_collocated_nodes = (dg_input->all_parameters->flux_nodes_type==Parameters::AllParameters::FluxNodes::GLL) && (dg_input->all_parameters->overintegration==0);
-            using PDEEnum = Parameters::AllParameters::PartialDifferentialEquation;
-            const PDEEnum pde_type = dg_input->all_parameters->pde_type;
-            const bool use_inviscid_burgers = (pde_type == PDEEnum::burgers_inviscid);
-            if (use_collocated_nodes && use_inviscid_burgers){
-                pcout << "Creating Relaxation Runge Kutta ODE Solver with " 
-                      << n_rk_stages << " stage(s)..." << std::endl;
-                if (n_rk_stages == 1){
-                    return std::make_shared<RRKExplicitODESolver<dim,real,1,MeshType>>(dg_input,rk_tableau);
-                }
-                if (n_rk_stages == 2){
-                    return std::make_shared<RRKExplicitODESolver<dim,real,2,MeshType>>(dg_input,rk_tableau);
-                }
-                if (n_rk_stages == 3){
-                    return std::make_shared<RRKExplicitODESolver<dim,real,3,MeshType>>(dg_input,rk_tableau);
-                }
-                if (n_rk_stages == 4){
-                    return std::make_shared<RRKExplicitODESolver<dim,real,4,MeshType>>(dg_input,rk_tableau);
-                }
-                else{
-                    pcout << "Error: invalid number of stages. Aborting..." << std::endl;
-                    std::abort();
-                    return nullptr;
-                }
-            } else {
-                pcout << "Error: RRK has only been tested on collocated nodes with Burgers. Aborting..."<<std::endl;
-                std::abort();
-            }
-        } else {
-            pcout << "Error: RRK has only been tested on 1D calculations. Aborting..."<<std::endl;
+
+        using PDEEnum = Parameters::AllParameters::PartialDifferentialEquation;
+        const PDEEnum pde_type = dg_input->all_parameters->pde_type;
+        using NumFluxEnum = Parameters::AllParameters::TwoPointNumericalFlux;
+        const NumFluxEnum two_point_num_flux_type = dg_input->all_parameters->two_point_num_flux_type;
+        
+        enum NumEntropyEnum {energy, entropy};
+        NumEntropyEnum numerical_entropy_type;
+        std::string numerical_entropy_string;
+        if (pde_type == PDEEnum::burgers_inviscid){
+            numerical_entropy_type = NumEntropyEnum::energy;
+            numerical_entropy_string = "Energy";
```

Comment:
This implementation differentiates between the energy (algebraic) RRK formulation of Ketcheson and the entropy (root-finding) RRK formulation of Ranocha. I'm using the PDE type as an indicator for this -- the block starting at line 171 chooses either an `EnergyRRKODESolver` or `EntropyRRKODESolver`.

Do you think that a different naming convention would be clearer here? I agree that it may be confusing to differentiate numerical entropy as "energy" or "entropy" here.

Reply:
Understood. Just to restate what you have expressed, your energy denotes an algebraic approach to computing the relaxation approach; while entropy is through a root-finding approach. What if we wanted to do kinetic_energy_preserving? I guess, how we compute the relaxation parameter would not matter since it depends on whether the resulting equation is linear in \gamma or not.

Reply:
That's correct. Any general nonlinear numerical entropy variable would use the "entropy" formulation. I suppose that kinetic energy preserving would also need to use the root-finding formulation.

If energy <u,u> is the numerical entropy, then the algebraic approach is used.

Reply:
I think a different naming convention would be easier to follow but its a minor point.

Reply:
I could rename to "AlgebraicRRK" and "RootRRK".